stellargraph/connector/neo4j/sampler.py

```diff
@@ -53,16 +53,20 @@ def _bfs_neighbor_query(sampling_direction, id_property, node_label=None):
             // find the neighbors
             MATCH (cur_node){direction_arrow}(neighbors)
 
-            // put all ids into a list
-            WITH CASE collect(neighbors.{id_property}) WHEN [] THEN [null] ELSE collect(neighbors.{id_property}) END AS in_neighbors_list
+            // collect neighbors in a list
+            WITH CASE collect(neighbors) WHEN [] THEN [null] ELSE collect(neighbors) END AS in_neighbors_list
 
             // pick random nodes with replacement
-            WITH apoc.coll.randomItems(in_neighbors_list, $num_samples, True) AS in_samples_list
+            WITH apoc.coll.randomItems(in_neighbors_list, $num_samples, True) AS sampled
+
+            // pull the ids of the sampled nodes only
+            unwind sampled as nn
+            WITH CASE collect(nn.{id_property}) WHEN [] THEN sampled ELSE collect(nn.{id_property}) END AS in_samples_list
```

Comment:
I've lost the context for this code, so could you remind me why this needs to do the same `WITH CASE ...` code as the `neighbors` one above? Does something like the following work?

```suggestion
            WITH collect(nn.{id_property}) AS in_samples_list
```

Or even

```suggestion
            RETURN collect(nn.{id_property}) AS in_samples_list
```

and delete the `RETURN in_samples_list` below.

Reply:
the problem with this is that when every thing in `sampled` is null this will return a singleton list `[ null ]` rather than what you need down stream which is list that is `num_samples` long.

Reply:
Ah, cool. I didn't realise that [collect()](https://neo4j.com/docs/cypher-manual/current/functions/aggregating/#functions-collect) ignores `null`, which makes this all make sense!

Arguably one could very clean it up with an additional `WITH`, but I don't think that's necessary (this also optionally applies to the `collect(neighbors)` code above).

```suggestion
            WITH collect(nn.{id_property}) AS in_samples_list
            // collect ignores nulls, so re-handle the no-neighbours case to ensure we get [null, null, ...] output
            WITH CASE in_samples_list WHEN [] THEN sampled ELSE in_samples_list END AS in_samples_list
```